homework/shared_ptr/weak_ptr.hpp

```diff
@@ -89,4 +84,23 @@ weak_ptr<T>& weak_ptr<T>::operator=(weak_ptr<T>&& previousOwner) noexcept {
     return *this;
 }
 
+template <typename T>
+void weak_ptr<T>::reset() noexcept {
+    if (!counter_->getWeakRefs()) {
+        delete ptr_;
+    } else {
```

Comment:
weak_ptr does not manage that ptr_. All that reset is suppose to do is make weak_ptr stop being associated with it.

Reply:
a so object member in weak_ptr should be set to nullptr as i understand

Reply:
yes, ptr should be set to nullptr, counter should be first decreased and then set to nullptr

Reply:
done